src/pages/index.js

```diff
@@ -0,0 +1,292 @@
+const avatarModal = document.querySelector("#edit-avatar-modal");
+const previewModal = document.querySelector("#view-photo");
+const deleteModal = document.querySelector("#delete-card-modal");
+const closeButtons = document.querySelectorAll(".modal__close-btn");
+
+// Forms and inputs
+const profileForm = document.forms["edit-profile-form"];
+const nameInput = document.querySelector("#name-input");
+const descriptionInput = document.querySelector("#description-input");
+const cardForm = document.forms["new-post-form"];
+const cardNameInput = newPostModal.querySelector("#caption-input");
+const cardLinkInput = newPostModal.querySelector("#image-link");
+const avatarForm = document.forms["edit-avatar"];
+const avatarLinkInput = document.querySelector("#avatar-link");
+const deleteForm = deleteModal.querySelector("form");
+const deleteButton = deleteForm.querySelector('button[type="submit"]');
+const cancelButton = deleteForm.querySelector('button[type="button"]');
+
+// Form submit buttons
+const avatarFormSubmitButton = avatarForm.querySelector(".modal__submit-btn");
+const cardFormSubmitButton = cardForm.querySelector(".modal__submit-btn");
+const profileFormSubmitButton = profileForm.querySelector(".modal__submit-btn");
+
+// Card elements
+const cardTemplate = document.querySelector("#card-template").content;
+const cardList = document.querySelector(".gallery__cards");
+const previewModalImageEl = previewModal.querySelector(".modal__image");
+const previewModalCaptionEl = previewModal.querySelector(".modal__caption");
+
+// Store the current selected card and its ID
+let selectedCard;
+let selectedCardId;
+
+api
+  .getAppInfo()
+  .then(([cardsData, userData]) => {
+    avatarImage.src = userData.avatar;
+    profileNameElement.textContent = userData.name;
+    profileDescriptionElement.textContent = userData.about;
+
+    cardsData.forEach((cardData) => {
+      const cardElement = getCardElement(cardData);
+      cardList.append(cardElement);
+    });
+  })
+  .catch(console.error);
+
+// Opens a modal
+function openModal(modal) {
+  modal.classList.add("modal_opened");
+  document.addEventListener("keydown", handleEscClose);
+}
+
+// Closes a modal
+function closeModal(modal) {
+  modal.classList.remove("modal_opened");
+  document.removeEventListener("keydown", handleEscClose);
+}
+
+// Opens profile edit modal
+function openEditProfileModal() {
+  nameInput.value = profileNameElement.textContent;
+  descriptionInput.value = profileDescriptionElement.textContent;
+  resetValidation(profileForm, [nameInput, descriptionInput], settings);
+  openModal(editProfileModal);
+  disableButton(profileFormSubmitButton, settings);
+}
+
+// Opens new post modal
+function openNewPostModal() {
+  openModal(newPostModal);
+}
+
+// Opens avatar edit modal
+function openAvatarModal() {
+  resetValidation(avatarForm, [avatarLinkInput], settings);
+  openModal(avatarModal);
+  disableButton(avatarFormSubmitButton, settings);
```

Comment:
You need to `disable` the button only after **resetting** the form when you have submitted a card otherwise users may close the modal by accident and they don’t expect that the button will be disabled on opening if the inputs have **valid** data which they haven’t submitted yet.

You need to **disable** the button only in a `then` block after a **successful response** to be able to click it again if there is a server error.